### ISSUE TITLE
Prometheus v2.6.0-rc.0 changes container args, breaking prometheus-operator deployments

### ISSUE THREAD

**Issue by matthiaslee** [coreos/prometheus-operator#2201]
Prometheus v2.6.0-rc.0 was cut yesterday, included in this release are changes ([#4796](https://github.com/prometheus/prometheus/pull/4796/files)) which change the `CMD` and `ENTRYPOINT` of the prometheus container.

![image](https://user-images.githubusercontent.com/1522304/49600931-76c42280-f952-11e8-96d1-ba87cf725a66.png)

Since the `prometheus-operator` overrides the CMD ([prometheus/statefulset.go](https://github.com/coreos/prometheus-operator/blob/v0.26.0/pkg/prometheus/statefulset.go#L776), [K8s 1.13 Container API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#container-v1-core)) and not the ENTRYPOINT, we end up with two copies of the `--config.file=` argument. This causes prometheus to crash with the following error message:
```
Error parsing commandline arguments: flag 'config.file' cannot be repeated
```

**Comment by brancz** [comment]
Thanks for reporting this! This should be a simple fix. If we're already at it, let's add 2.6.0-rc.0 to our compatibility matrix to ensure it works.

cc @metalmatze @squat @s-urbaniak @mxinden

**Comment by simonpasquier** [comment]
We have found a way to improve the packaging container image while not breaking existing deployments. See https://github.com/prometheus/prometheus/pull/4976

**Comment by matthiaslee** [comment]
@brancz should we close this then?

**Comment by simonpasquier** [comment]
Probably best to wait until https://github.com/prometheus/prometheus/pull/4976 is merged.

**Comment by metalmatze** [comment]
Very nice, thank you for the heads up! I had actually planned to work on this today, but thankfully got distracted, so it seems.

**Comment by simonpasquier** [comment]
The upstream PR is merged. We plan to cut the RC.1 on Wednesday to take advantage of the security bug fix in the upcoming [Go 1.11.3](https://groups.google.com/d/msg/golang-dev/Yk09_1ktph8/KtNl1OAZBgAJ).

**Comment by matthiaslee** [comment]
@simonpasquier thanks for the heads up. I'll close this now.
